src/sinks/aws_s3/sink.rs

```diff
@@ -0,0 +1,521 @@
+                    .instrument(info_span!("request", request_id = %seqno));
+                tokio::spawn(fut);
+            },
+
+            Some(Ok((seqno, batch_size))) = in_flight.next() => {
+                trace!("pending batch {} finished (n={})", seqno, batch_size);
+                pending_acks.insert(seqno, batch_size);
+
+                let mut num_to_ack = 0;
+                while let Some(ack_size) = pending_acks.remove(&seq_tail) {
+                    num_to_ack += ack_size;
+                    seq_tail += 1
+                }
+                trace!(message = "Acking events.", acking_num = num_to_ack);
+                acker.ack(num_to_ack);
+            },
+
+            else => break
+        }
+    }
+}
+
+fn build_request(key: String, batch: Vec<Event>, options: &S3RequestOptions) -> S3Request {
+    // Generate the filename for this batch, which involves a surprising amount
+    // of code.
+    let filename = {
+        /*
+        Since this is generic over the partitioner, for purposes of unit tests,
+        we can't get the compiler to let us define a conversion trait such that
+        we can get &Event from &P::Item, or I at least don't know how to
+        trivially do that.  I'm leaving this snippet here because it embodies
+        the prior TODO comment of using the timestamp of the last event in the
+        batch rather than the current time.
+
+        Now that I think of it... is that even right?  Do customers want logs
+        with timestamps in them related to the last event contained within, or
+        do they want timestamps that include when the file was generated and
+        dropped into the bucket?  My gut says "time when the log dropped" but
+        maybe not...
+
+        let last_event_ts = batch
+            .items()
+            .iter()
+            .last()
+            .and_then(|e| match e.into() {
+                // If the event has a field called timestamp, in RFC3339 format, use that.
+                Event::Log(le) => le
+                    .get(log_schema().timestamp_key())
+                    .cloned()
+                    .and_then(|ts| match ts {
+                        Value::Timestamp(ts) => Some(ts),
+                        Value::Bytes(buf) => std::str::from_utf8(&buf)
+                            .ok()
+                            .and_then(|s| DateTime::parse_from_rfc3339(s).ok())
+                            .map(|dt| dt.with_timezone(&Utc)),
+                        _ => None,
+                    }),
+                // TODO: We don't ship metrics to the S3, but if we did, would this be right? or is
+                // there an actual field we should be checking similar to above?
+                Event::Metric(_) => Some(Utc::now()),
+            })
+            .unwrap_or_else(|| Utc::now());
+        let formatted_ts = last_event_ts.format(&time_format);
+        */
+        let formatted_ts = Utc::now().format(options.filename_time_format.as_str());
```

Comment:
This seems very reasonable and likely preferable to the old commented-out logic above. I'd say we document the change and delete the old version above.

Reply:
Agreed. Where do we have the current behavior documented?

Reply:
It looks like we're not very specific currently. Source is [here](https://github.com/timberio/vector/blob/f1132d045b4ed0bcedc8983bc775af98bcba3b6d/website/cue/reference/components/sinks/aws_s3.cue#L323-L356). Likely worth specifying that it's the time the request is made and maybe dropping a note in the changelog. The only case I can think of where this would be a meaningful change is when working with historical data or similar.

Reply:
I think, if your amenable to the notion, I'd like to make a follow-up issue for this. We should have a test in place that enforces documented behavior, which we don't right now.